Bug 1146863, improve partial suggestions by using more accurate data source

nthomas-mozilla opened the pull request:
This swaps from an unloved FHR data source in Vertica back to blocklist ADI via crash-stats (who have a nice API explorer at https://crash-stats.mozilla.com/api/).  In the long run we want to be using a fast FHR source (bug 1240522) but there's no heat behind that right now. In the meantime ship-it keeps making bad suggestions for release candidates, and sometimes need to rebuild for that.

There is an open question - when we're doing early betas, does the ship-it UI figure out the last X.0buildN when we translate X.0b99 to X.0. I could test this with a database dump if that's easy, or you know, in production. 😃 

WebOps will need to be involved in the deploy to adjust the virtualenv and cron job.

rail commented on scripts/update-ADI.py:
```diff
@@ -0,0 +1,61 @@
+    with open(exportFile, "w") as outfile:
+        json.dump(full, outfile)
+
+saveAllPartial("partial.json")
```
A nit. Can you wrap this in "if __name__ == '__main__'"?

rail commented on scripts/vertica-config.ini.template:
```diff
@@ -1,7 +0,0 @@
-host = <hostname.mozilla.com>
-port = 5433
-user = <login>
-password = <pass>
```
sweeet!!!! no more credentials!

rail commented on scripts/update-ADI.py:
```diff
@@ -0,0 +1,61 @@
+                                           timestamp.strftime("%d/%m/%Y"))
+    full = {"beta": partialBeta, "release": partialRelease, "esr": partialESR, "lastUpdate": lastUpdate}
+
+    with open(exportFile, "w") as outfile:
```
Would me safer to write to a temp file and then rename it, but meh.

rail commented:
:shipit:

nthomas-mozilla commented:
I've added requests to the vendor library (with `pip install -I --no-binary :any: --prefix $(PWD)/vendor --no-deps requests==2.10.0`, then manual maniuplation as I couldn't get pip to do the right thing re leaving out site-packages/ and python/ vs python2.7). Also added a symlink so that the existing cron job keeps working until webops have time to modify puppet.
